components/CreateReplyRequestForm/CreateReplyRequestForm.js

```diff
@@ -139,11 +139,18 @@ const SubmitButton = ({
 };
 
 const CreateReplyRequestForm = React.memo(
-  ({ articleId, articleUserId, requestedForReply, onNewReplyButtonClick }) => {
+  ({
+    articleId,
+    articleUserId,
+    requestedForReply,
+    replyRequest,
+    onNewReplyButtonClick,
+  }) => {
     const buttonRef = useRef(null);
     const [disabled, setDisabled] = useState(false);
     const [showForm, setShowForm] = useState(false);
     const [showFloatButton, setShowFloatButton] = useState(false);
+    const [isRequestedForReply, setIsRequestedForReply] = useState(false);
```

Comment:
After testing on my local machine, I. can see that why you added `isRequestedForReply` -- it seems that `requestedForReply` boolean prop does not work properly. However, the current `useState` + `useEffect` implementation does not help displaying button text after the user refresh the page:
<img width="857" alt="image" src="https://user-images.githubusercontent.com/108608/230834811-0e38096b-e0c0-427f-9ecd-a50c8073bb13.png">

Since we have a new prop `replyRequest` now, I believe we can determine the value of `isRequestedForReply` directly from `replyRequest`, so the newly added `useState` and `useEffect` is probably not needed at all.

I would suggest
1. Remove `requestedForReply` prop from `CreateReplyRequestForm ` -- it's broken, and its purpose can be replaced by the new prop `replyRequest`
2. We either change `isRequestedForReply` to a boolean const that is calculated from `replyRequest`, or we just drop `isRequestedForReply` as well, because it's only used in one placed and its logic is simple enough to do it inline

Reply:
ok, i will try to fix.